edge/pkg/metamanager/client/configmap.go

```diff
@@ -58,15 +58,9 @@ func (c *configMaps) Get(name string) (*api.ConfigMap, error) {
 		return nil, fmt.Errorf("get configmap from metaManager failed, err: %v", err)
 	}
 
-	var content []byte
-	switch msg.Content.(type) {
-	case []byte:
-		content = msg.GetContent().([]byte)
-	default:
-		content, err = json.Marshal(msg.Content)
-		if err != nil {
-			return nil, fmt.Errorf("marshal message to configmap failed, err: %v", err)
-		}
+	content, err := msg.GetContentData()
+	if err != nil {
+		return nil, fmt.Errorf("marshal message to configmap failed, err: %v", err)
```

Comment:
 For `GetContentData` that return an error containing `marshal message content failed`.
https://github.com/kubeedge/kubeedge/blob/c51143389cdd3061a804b73be21b951c446bb68a/staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go#L142-L152

I recommend avoiding including marshal-related information in an error

Reply:
done, modify to `parse message to configmap failed`